packages/react/src/hooks/useConversationWith.ts

```diff
@@ -0,0 +1,384 @@
+      ConnectionState.Connected, /* FIXME: should I check for other states too? */
+      signal,
+    );
+  }, [waitUntilConnectionState]);
+
+  const waitUntilDisconnected = useCallback(async (signal?: AbortSignal) => {
+    return waitUntilConnectionState(
+      ConnectionState.Disconnected,
+      signal,
+    );
+  }, [waitUntilConnectionState]);
+
+  const agent = useAgent(useMemo(() => ({
+    connectionState: conversationState.connectionState,
+    subtle: {
+      emitter,
+      room,
+      credentials: options.tokenSource,
+      agentConnectTimeoutMilliseconds: options.dispatch?.agentConnectTimeoutMilliseconds,
+    },
+  }), [conversationState, emitter, room, options.tokenSource, options.dispatch?.agentConnectTimeoutMilliseconds]), agentName);
+
+  const start = useCallback(async (connectOptions: AgentSessionConnectOptions = {}) => {
+    const {
+      signal,
+      tracks = { microphone: { enabled: true, publishOptions: { preConnectBuffer: true } } },
+    } = connectOptions;
+
+    await waitUntilDisconnected(signal);
+
+    const onSignalAbort = () => {
+      room.disconnect();
+    };
+    signal?.addEventListener('abort', onSignalAbort);
+
+    await Promise.all([
+      // FIXME: swap the below line in once the new `livekit-client` changes are published
+      // room.connect(options.credentials),
+      options.tokenSource.generate().then(({ serverUrl, participantToken }) => (
+        room.connect(serverUrl, participantToken)
+      )),
+
+      // Start microphone (with preconnect buffer) by default
+      tracks.microphone?.enabled ? (
+        room.localParticipant.setMicrophoneEnabled(true, undefined, tracks.microphone?.publishOptions ?? {})
+      ) : Promise.resolve(),
+    ]);
+
+    await waitUntilConnected(signal);
+    await agent.waitUntilAvailable(signal);
+
+    signal?.removeEventListener('abort', onSignalAbort);
+  }, [room, waitUntilDisconnected, options.tokenSource, waitUntilConnected, agent.waitUntilAvailable]);
+
+  const end = useCallback(async () => {
+    await room.disconnect();
+  }, [room]);
+
+  const prepareConnection = useCallback(async () => {
+    const credentials = await options.tokenSource.generate();
+    await room.prepareConnection(credentials.serverUrl, credentials.participantToken);
+  }, [options.tokenSource, room]);
+  useEffect(() => {
+    prepareConnection().catch(err => {
+      // FIXME: figure out a better logging solution?
+      console.warn('WARNING: Room.prepareConnection failed:', err);
+    });
+  }, [prepareConnection]);
+
+  return useMemo(() => ({
+    ...conversationState,
+
+    waitUntilConnected,
+    waitUntilDisconnected,
+
+    prepareConnection,
+    start,
+    end,
+  }), [
+    conversationState,
+    waitUntilConnected,
+    waitUntilDisconnected,
+    prepareConnection,
+    start,
+    end,
+  ]);
+}
+
+export function useConversation(options: ConversationOptions): ConversationInstance {
+  return useConversationWith(null, options);
+}
```

Comment:
Right now, there are two forms of this hook - `useConversationWith("agent name to dispatch", { /* options */})` and `useConversation({ /* options */})`. The later exists to support automatic agent dispatch.

Does the agents sdk need to support automatic agent dispatch?  It is unclear if this is something that is important or not for the agents sdk to support since what I can gather, this is being phased out.